src/EditorFeatures/TestUtilities/Threading/WpfTestSharedData.cs

```diff
@@ -13,21 +13,13 @@ public sealed class WpfTestSharedData
     {
         internal static readonly WpfTestSharedData Instance = new WpfTestSharedData();
 
-        /// <summary>
-        /// The name of a <see cref="Semaphore"/> used to ensure that only a single
-        /// <see cref="WpfFactAttribute"/>-attributed test runs at once. This requirement must be made because,
-        /// currently, <see cref="WpfTestCase"/>'s logic sets various static state before a method runs. If two tests
-        /// run interleaved on the same scheduler (i.e. if one yields with an await) then all bets are off.
-        /// </summary>
-        internal static readonly Guid TestSerializationGateName = Guid.NewGuid();
```

Comment:
📝 If we are getting rid of this, we can also make `WpfTestSharedData` no longer serializable and just have code access `WpfTestSharedData.Instance` directly.

Reply:
Sure, I'll remove the serializable attribute since yeah there's no way a SemaphoreSlim is serializable in the first place.

I don't understand your second comment though: it seemed that all the code *is* just accessing it through the Instance property already. Even cases where we're in Deserialization methods it seems we just didn't deserialize it and re-used the instance anyways like here:

https://github.com/dotnet/roslyn/blob/1da3e28e3b99ea329262878d064ba4debc4db70f/src/EditorFeatures/TestUtilities/Threading/WpfTheoryTestCase.cs#L26-L30

Reply:
I'm not sure you can safely remove the serialization code. All of that logic is just boiler plate copied from xunit. Whenever it was wrong it failed a random unit scenario. If you change any of it make sure you do a test run where you pass the `-serialize` argument to xunit. That will force the issue and validate everything is correct.

Reply:
So the [Serializable] attribute on WpfTestSharedData I suspect wasn't doing anything for a couple of reasons:

1. Semaphores aren't serializable, so WpfTestSharedData couldn't have been serialized.
2. Nothing ever actually use a serialized instance that I can find.
3. In practice due to the static initialization order I'm not confident what would have been serialized correctly -- if serialization ran after the initial instance it might have serialized over the set Guid name, which wouldn't have worked.

Per [this comment](https://github.com/xunit/xunit/issues/1353#issuecomment-335018203) from @bradwilson, no matter what we're still following the real requirement that we implement IXunitSerializable, which we're doing because our base type already does, and we're delegating to our base type. I'll clean it up a bit further.

Reply:
I should also mention there's a lot of other code smells in this area too that generally make me think this was all refactored and this bit just wasn't cleaned up. For example this:

https://github.com/dotnet/roslyn/blob/1da3e28e3b99ea329262878d064ba4debc4db70f/src/EditorFeatures/TestUtilities/Threading/WpfTestSharedData.cs#L24-L28

Says it's the only the last 10 test cases. Except there's no clearing anywhere, so we hold onto the entire list regardless. :smile:

Reply:
@sharwell Added 1b87cb1f4145fe6eab751cc0e555387f48899ab to remove the `[Serializable]` and cleaned it up a bit further. Tested with `-serialize` and confirmed it still works.